index.ts

```diff
@@ -93,7 +108,22 @@ async function handleMessage(
   return { ...response, __webext_messenger__ };
 }
 
-async function handleCall(
+// Do not turn this into an `async` function; Notifications must turn `void`
+function manageConnection(
```

Comment:
Names are a bit messy at the moment. This is because there are a few helper functions that just aim at reducing duplication, but are meant to handle as much of the logic as possible.

They will likely disappear once I manage to merge `getContentScriptMethod` into `getMethod`. The reason why they're not merged now are the types, which just grew 3-fold with this PR due to overloads required by notifications. 😬 

No need to clean this up now because it's all transitional.